src/js/send-web-vitals.js

```diff
@@ -2,26 +2,50 @@ function sendWebVitals() {
 
   function sendWebVitalsGAEvents({name, delta, id, attribution, navigationType}) {
 
-    let webVitalInfo = '(not set)';
+    let overrides = {};
 
     switch (name) {
       case 'CLS':
-        webVitalInfo = attribution.largestShiftTarget;
+        overrides = {
+          debug_time: attribution.largestShiftTime,
+          debug_load_state: attribution.loadState,
```

Comment:
This is repeated for a few metrics. Should it be pulled into the "default" object below?

Reply:
Good spot but don't think it makes much difference so would prefer not to set it on the two metrics it's not set for (TTFB and LCP).

Reply:
Why wouldn't it be useful for TTFB and LCP? FWIW I think the distinction you've started drawing between metric-specific diagnostics and general diagnostics is good, and this falls more cleanly under the general category.

Reply:
Well it's more that it's not available for them [in the web-vitals attribution model](https://github.com/GoogleChrome/web-vitals#attribution).

For TTFB it doesn't really make sense (that happens before any load state). For LCP maybe it should be added to web-vitals? Though that's a discussion for that repo rather than this one.

The web vitals library does give reasons why to measure them for most of the other 4 (CLS being the exception as it's a little vague for that one):
- FCP - "Ideally, documents can paint before they finish loading (e.g. the `loading` or `dom-interactive` phases)."
- INP/FID - "If the first interaction occurred while the document was loading and executing script (e.g. usually in the  `dom-interactive` phase) it can result in long input delays."

Don't think those really apply to LCP.

Reply:
Fair enough. We can conditionally set it, similar to the user preferences, and it'll be null for metrics that don't apply. I'll leave it up to you.